cmd/sync/plus/nginx_api.go

```diff
@@ -0,0 +1,88 @@
+package plus
+
+import (
+	"log"
+	"net/http"
+)
+
+// NginxAPIController stores a NginxClient.
+type NginxAPIController struct {
+	client *NginxClient
+}
+
+// ServerConfig stores a server configuration.
+type ServerConfig struct {
+	MaxFails    int
+	FailTimeout string
+}
+
+// NewNginxAPIController creates an NginxClient and returns a NginxAPIController.
+func NewNginxAPIController(httpClient *http.Client, endpoint string) (*NginxAPIController, error) {
+	client, err := NewNginxClient(httpClient, endpoint)
+
+	if err != nil {
+		return nil, err
+	}
+	nginx := &NginxAPIController{client: client}
+	return nginx, nil
+}
+
+// UpdateServers updates the HTTPServers of the upstream.
+// HTTP Servers that are in the slice, but don't exist in NGINX, will be added to NGINX.
+// HTTP Servers that are not in the slice, but exist in NIGNX, will be removed from NGINX.
+func (nginx *NginxAPIController) UpdateServers(upstream string, servers []string, config ServerConfig) error {
+	var upsServers []UpstreamServer
```

Comment:
move this logic  - lines 34-41 into main
from main, call UpdateHTTPServers of NginxClient

Reply:
also, set `MaxFails` to `1`, as it is NGINX default

Reply:
same applies to UpdateStreamServers